src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaderValueCollection.cs

```diff
@@ -97,18 +97,32 @@ public void CopyTo(T[] array!!, int arrayIndex)
 
             if (storeValues == null)
             {
-                // We only have 1 value: If it is the "special value" just return, otherwise add the value to the
-                // array and return.
-                Debug.Assert(storeValue is T);
-                if (arrayIndex == array.Length)
+                if (storeValue is not HttpHeaders.InvalidValue)
                 {
-                    throw new ArgumentException(SR.net_http_copyto_array_too_small);
+                    // We only have 1 value: If it is the "special value" just return, otherwise add the value to the
```

Comment:
This comment about "special value" was incorrect previously. Rather than leave it, let's just remove the comment.